layout: Do not use orthogonal baselines in flex layout

servo-wpt-sync opened the pull request:
When a baseline is orthogonal to the main flexbox axis, it should not
take part in baseline alignment. This change does that for column flex.
While there is no support for vertical writing modes, this change is
made to be as writing mode-agnostic as possible.

Signed-off-by: Martin Robinson <mrobinson@igalia.com>
Co-authored-by: Oriol Brufau <obrufau@igalia.com>

<!-- Please describe your changes on the following line: -->


Reviewed in servo/servo#33347

bfgeek commented on css/css-flexbox/flexbox-align-self-baseline-compatability.html:
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE>
```
@mrobinson @Loirooriol  quirky doc-type causing failures in other engines.

mrobinson replied:
Thanks for the heads up. This should be fixed now.

Loirooriol commented on css/css-flexbox/flexbox-align-self-baseline-compatability.html:
```diff
@@ -0,0 +1,36 @@
+            width: 100px;
+            height: 50px;
+            background: green;
+            color: green;
```
The "A" is overflowing a little bit on Firefox. better use `color: transparent`.

mrobinson committed:
layout: Do not use orthogonal baselines in flex layout

When a baseline is orthogonal to the main flexbox axis, it should not
take part in baseline alignment. This change does that for column flex.
While there is no support for vertical writing modes, this change is
made to be as writing mode-agnostic as possible.

Signed-off-by: Martin Robinson <mrobinson@igalia.com>
Co-authored-by: Oriol Brufau <obrufau@igalia.com>
